src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -16,15 +16,15 @@ internal partial class BoundExpression
         /// </summary>
         protected abstract BoundExpression ShallowClone();
 
-        internal BoundExpression WithSuppression()
+        internal BoundExpression WithSuppression(bool suppress = true)
         {
-            if (this.IsSuppressed)
+            if (this.IsSuppressed == suppress)
             {
                 return this;
             }
 
             var result = ShallowClone();
-            result.IsSuppressed = true;
+            result.IsSuppressed = suppress;
```

Comment:
>result.IsSuppressed = suppress; [](start = 12, length = 31)

There should be no scenario where we go from "suppressed" to "not suppressed". Is that worth asserting?